Source/EasyNetQ/Consumer/HandlerCollection.cs

```diff
@@ -71,13 +67,15 @@ public HandlerCollection(IEasyNetQLogger logger)
             return (message, info) => Task.Factory.StartNew(() => { });
         }
 
-        public dynamic GetHandler(Type messageType)
+        public Func<IMessage<object>, MessageReceivedInfo, Task> GetHandler(Type messageType)
         {
             Preconditions.CheckNotNull(messageType, "messageType");
-
+            
             var getHandlerGenericMethod = GetType().GetMethod("GetHandler", new Type[0]).MakeGenericMethod(messageType);
+            
+            var func = (Delegate)getHandlerGenericMethod.Invoke(this, new object[0]);
```

Comment:
I took that code from previous one, but you're right, I don't think is need it, I'll fix it.